lib/excon/connection.rb

```diff
@@ -72,21 +73,21 @@ def initialize(params = {})
       # Use Basic Auth if url contains a login
       if @data[:user] || @data[:password]
         user, pass = Utils.unescape_form(@data[:user].to_s), Utils.unescape_form(@data[:password].to_s)
-        @data[:headers]['Authorization'] ||= 'Basic ' << ['' << user.to_s << ':' << pass.to_s].pack('m').delete(Excon::CR_NL)
+        @data[:headers]['Authorization'] ||= 'Basic ' + ["#{user}:#{pass}"].pack('m').delete(Excon::CR_NL)
       end
 
-      @socket_key = '' << @data[:scheme]
+      @socket_key = @data[:scheme].dup
```

Comment:
I think we can probably do away with this line now to simplify and just have a single interpolating line below. What do you think?